jobs/healing/pipeline-healing.js

```diff
@@ -229,23 +232,91 @@ async function getScopedSourceTriples( config, property, conceptSchemeUri, publi
   return reformatQueryResult(result);
 }
 
+function arrayToFile(array, file){
+  let fd = fs.openSync(file, "w");
+  for (let i=0; i<array.length; ++i){
+    fs.writeSync(fd, JSON.stringify(array[i]) + "\n")
+  }
+  fs.close(fd)
+  return file
+}
+function getTempFile() {
+  let file = "/tmp/healing-" + uuid();
+  return file;
+}
+function lines(filename) {
+  let retval = []
+  let rl = new readlines(filename)
+  let line
+  while ((line = rl.next())) {
+    line = line.toString()
+    if (line[0] === "<" || line[0] === ">") line = line.replace(/^./, "");
+    retval.push(JSON.parse(line) || {})
+  }
+  return retval;
+}
+function diffFiles(targetFile, sourceFile, S="50%", T="/tmp"){
+  // Note: the S and T parameters can be used to tweak the memory usage of the sort command
+  const optionsNoOutput = {
+    encoding: 'utf-8',
+    stdio: ['ignore', 'ignore', 'ignore']
+  }
+
+  let sorted1 = getTempFile();
+  let sorted2 = getTempFile();
+
+  execSync(`sort ${targetFile} -S ${S} -T ${T} -o ${sorted1}`, optionsNoOutput)
+  execSync(`sort ${sourceFile} -S ${S} -T ${T} -o ${sorted2}`, optionsNoOutput)
+
+  let output1 = getTempFile();
+  let output2 = getTempFile();
+
+  execSync(`comm -23 ${sorted1} ${sorted2} | tee ${output1}`)
+  execSync(`comm -13 ${sorted1} ${sorted2} | tee ${output2}`)
+
+  let inserts = lines(output1)
+  let deletes = lines(output2)
+
+  execSync(`rm -fv ${sorted1} ${sorted2} ${output1} ${output2}`, optionsNoOutput)
```

Comment:
is this something that can either be fixed by a lib, or try/catch/finally

Reply:
solved by using lib tmp